src/Processors/Transforms/Streaming/GlobalAggregatingTransformWithSubstream.cpp

```diff
@@ -54,11 +62,13 @@ GlobalAggregatingTransformWithSubstream::executeOrMergeColumns(Chunk & chunk, co
         assert(!params->only_merge && !no_more_keys);
 
         auto num_rows = chunk.getNumRows();
-        auto & retracted_variants = substream_ctx->getField<RetractedDataVariantsPtr>();
-        auto & aggregated_variants = substream_ctx->variants;
-
-        return params->aggregator.executeAndRetractOnBlock(
-            chunk.detachColumns(), 0, num_rows, aggregated_variants, *retracted_variants, key_columns, aggregate_columns);
+        auto retract_enabled = substream_ctx->getField<bool>();
```

Comment:
We can consider refactor out a readable inline function for `getField<bool>`  in this WithSubstream class, that will be more readable and maintainable. General idea is if we need cast void *, std::any and we will need do in different call sites, refactor this in a function which has a centralized places to handle the casting.

```
bool & retractEnabled() const noexcept
{
    return substream_ctx->getField<bool&>();
}
```